src/api.rs

```diff
@@ -77,78 +66,53 @@ impl<'a, T: CronCatInterface> CronCat<'a, T> {
     pub fn refill_task(
         &self,
         task_hash: String,
-        task_version: String,
         assets: AssetListUnchecked,
-    ) -> AbstractSdkResult<Vec<CosmosMsg>> {
-        let modules = self.base.modules(self.deps);
-        let querier = &self.deps.querier;
-        let (funds, cw20s) = sort_funds(self.deps, assets)?;
-        let mut messages = vec![];
-
-        let croncat_factory_address = modules.module_address(self.module_id)?;
-        let manager_addr = get_croncat_contract(
-            querier,
-            croncat_factory_address,
-            MANAGER_NAME.to_string(),
-            task_version,
-        )
-        .map_err(|e| AbstractSdkError::generic_err(e.to_string()))?;
-
-        // Note: It will be up to one cw20 for now
-        // but maybe later we can support multiple if we could confirm some "recipe" won't lock agents gas limit
-        for cw20 in cw20s {
-            let cw20_transfer = wasm_execute(
-                cw20.address,
-                &Cw20ExecuteMsg::Send {
-                    contract: manager_addr.to_string(),
-                    amount: cw20.amount,
-                    msg: to_binary(
-                        &croncat_sdk_manager::msg::ManagerReceiveMsg::RefillTaskBalance {
-                            task_hash: task_hash.clone(),
-                        },
-                    )?,
-                },
-                vec![],
-            )?;
-            messages.push(cw20_transfer.into());
-        }
-        if !funds.is_empty() {
-            let refill_msg = wasm_execute(
-                manager_addr,
-                &ManagerExecuteMsg::RefillTaskBalance {
-                    task_hash: task_hash.clone(),
-                },
-                funds,
-            )?;
-            messages.push(refill_msg.into());
-        }
-        Ok(messages)
+    ) -> AbstractSdkResult<CosmosMsg> {
+        Ok(wasm_execute(
+            self.module_address()?,
+            &AppExecuteMsg::RefillTask { task_hash, assets },
```

Comment:
For this message (and the query as well) you need to either turn it into an `ExecuteMsg` that is defined in [msg.rs](https://github.com/CyberHoward/abstract-croncat-app/pull/1/files/06d1585941faf9ec6d81ed53678351564dbecde4..b1cf5d6fad83f58b37bb918294f25b52212c34c1#diff-2a31c9bce208313e13c9ce0f229156f1f92090938ceb17950cacab5fa33e6dc7L10-L13). You can do that with a simple `ExecuteMsg::from(AppExecuteMsg::...)`. 

Or you can make use of the `Apps` API that allows you to execute and query on an app module without the need to convert the message yourself. You just provide the module ID and the message: https://docs.rs/abstract-sdk/latest/abstract_sdk/struct.Apps.html

Reply:
TYSM